app/controllers/solidus_square/payments_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  class PaymentsController < BaseController
+    def create
+      order = ::Spree::Order.find_by!(number: params[:payment][:order_number])
+      authorize! :update, order, order_token
+
+      SolidusSquare::CreatePaymentService.call(
+        source_id: params[:payment][:source_id],
+        order: order,
+        payment_method_id: params[:payment][:payment_method_id]
+      )
+
+      render json: {}, status: :ok
+    rescue StandardError => e
+      render json: { error: e.message }, status: :unprocessable_entity
+    end
+
+    private
+
+    def order_token
+      request.headers["X-Spree-Order-Token"]
+    end
+  end
```

Comment:
What do you think to reuse [that code](https://github.com/nebulab/solidus_square/blob/main/app/controllers/solidus_square/api/checkout_controller.rb#L24)?

Reply:
https://github.com/nebulab/solidus_square/pull/49/commits/eef3bb8aadb12106067bcb45c5eced8524348bb8